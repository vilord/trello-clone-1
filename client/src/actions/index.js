export const INFO_REQ = 'INFO_REQ';
export const TOGGLE_SHOW = 'TOGGLE_SHOW';
export const REQUEST_POSTS = 'REQUEST_POSTS';
export const RECEIVE_POSTS = 'RECEIVE_POSTS';

export const toggleShow = {
  type: TOGGLE_SHOW,
};

export const requestPosts = subreddit => ({
  type: REQUEST_POSTS,
  subreddit,
});

export const receivePosts = (subreddit, data) => ({
  type: RECEIVE_POSTS,
  subreddit,
  data,
});

/**
 * Async Actions
 */
export const fetchPosts = subreddit => {
  return dispatch => {
    dispatch(requestPosts(subreddit));
    return fetch(`https://www.reddit.com/r/${subreddit}.json`)
      .then(
        response => response.json(),
        error => console.log('An error occurred.', error),
      )
      .then(data => dispatch(receivePosts(subreddit, data)));
  };
};
