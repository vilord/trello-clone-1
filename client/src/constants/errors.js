export const invalidEmail = {
  kind: 'invalidEmail',
  header: 'Verify email!',
  message: "That doesn't look like an email addressâ€¦",
};

export const shortPassword = {
  kind: 'shortPassword',
  header: 'Password is too short!',
  message: 'Your password must be at least 8 characters long.',
};

export const userExists = {
  kind: 'userExists',
  header: 'User already exists!',
  message: 'Email is already in use by another account.'
};

export const wrongCredentials ={
  kind: 'wrongCredentials',
  header: 'Something wrong!',
  message: 'Incorrect email (username) or password.'
};

export const serverError = {
  kind: 'serverError',
  header: 'Unknown Server Error!',
  message: 'Our server may be down. Please retry later!'
};
