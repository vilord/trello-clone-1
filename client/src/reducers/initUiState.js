export default {
  activeMenus: {
    boardsExplorer: false,
    create: false,
    notifications: false,
    user: false,
    renameBoard: false,
    addToTeam: false,
    visibility: false,
    board: false,
  },
  search: '',
  // Focus options [ 'search' ]
  focus: '',
  fetching: {
    signup: false,
    login: false,
    activity: false,
  },
  error: {
    kind: '',
    header: '',
    message: '',
  },
};
