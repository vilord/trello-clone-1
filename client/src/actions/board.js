import * as types from '../constants/actionTypes';

export const setBoardTitle = title => ({
  type: types.SET_BOARD_TITLE,
  title,
});
