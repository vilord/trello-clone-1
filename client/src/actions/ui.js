import * as types from '../constants/actionTypes';

/**
 * UI Errors
 */
export const setUiError = error => ({
  type: types.SET_UI_ERROR,
  error,
});

export const resetUiError = () => ({
  type: types.RESET_UI_ERROR,
});

/**
 * Boards Explorer
 */
export const showBoardsExplorer = () => ({
  type: types.SHOW_BOARDS_EXPLORER,
});

export const hideBoardsExplorer = () => ({
  type: types.HIDE_BOARDS_EXPLORER,
});

/**
 * Header Search
 */
export const focusHeaderSearch = () => ({
  type: types.FOCUS_HEADER_SEARCH,
});

export const blurHeaderSearch = () => ({
  type: types.BLUR_HEADER_SEARCH,
});

export const setHeaderSearch = term => ({
  type: types.SET_HEADER_SEARCH,
  term,
});

/**
 * Create Menu
 */
export const showCreateMenu = () => ({
  type: types.SHOW_CREATE_MENU,
});

export const hideCreateMenu = () => ({
  type: types.HIDE_CREATE_MENU,
});

/**
 * Notifications
 */
export const showNotifications = () => ({
  type: types.SHOW_NOTIFICATIONS,
});

export const hideNotifications = () => ({
  type: types.HIDE_NOTIFICATIONS,
});

/**
 * User Menu
 */
export const showUserMenu = () => ({
  type: types.SHOW_USER_MENU,
});

export const hideUserMenu = () => ({
  type: types.HIDE_USER_MENU,
});

/**
 * Rename Board
 */
export const showRenameBoard = () => ({
  type: types.SHOW_RENAME_BOARD,
});

export const hideRenameBoard = () => ({
  type: types.HIDE_RENAME_BOARD,
});

/**
 * Add to Team
 */
export const showAddToTeam = () => ({
  type: types.SHOW_ADD_TO_TEAM,
});

export const hideAddToTeam = () => ({
  type: types.HIDE_ADD_TO_TEAM,
});

/**
 * Visibility Menu
 */
export const showVisibilityMenu = () => ({
  type: types.SHOW_VISIBILITY_MENU,
});

export const hideVisibilityMenu = () => ({
  type: types.HIDE_VISIBILITY_MENU,
});

/**
 * Board Menu
 */
export const showBoardMenu = () => ({
  type: types.SHOW_BOARD_MENU,
});

export const hideBoardMenu = () => ({
  type: types.HIDE_BOARD_MENU,
});
