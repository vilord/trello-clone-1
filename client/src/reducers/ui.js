import * as types from '../constants/actionTypes';

const initState = {
  // error: '',
  isFetching: false,
};

const uiReducer = (state = initState, action) => {
  switch (action.type) {
    case types.SIGNUP_USER_FAILURE:
      return {
        ...state,
        userExists: true,
      };
    case types.LOGIN_USER_REQUEST:
      return {
        ...state,
        isFetching: true,
      };
    case types.LOGIN_USER_SUCCESS:
      return {
        ...state,
        isFetching: false,
      };
    default:
      return state;
  }
};

export default uiReducer;
