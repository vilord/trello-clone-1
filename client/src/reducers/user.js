import * as types from '../constants/actionTypes';
import initUserState from './initUserState';

const userReducer = (state = initUserState, action) => {
  switch (action.type) {
    case types.SIGNUP_USER_SUCCESS:
      return {
        ...state,
        ...action.user,
      };
    case types.GET_USER_SESSION_SUCCESS:
    case types.LOGIN_USER_SUCCESS:
      return {
        ...state,
        ...action.user,
      };
    case types.SET_USER_PROFILE_SUCCESS:
      return {
        ...state,
        ...action.profile,
      };
    default:
      return state;
  }
};

/**
 * Higher order function that resets
 * the state to its initial value on
 * LOGOUT_USER action.
 */
const logoutWrapper = (state, action) => {
  if (action.type === types.LOGOUT_USER) {
    state = undefined;
  }
  return userReducer(state, action);
};

export default logoutWrapper;
