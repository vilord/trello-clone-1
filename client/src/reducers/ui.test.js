// import * as types from '../constants/actionTypes';
import * as uiActions from '../actions/ui';
import * as userActions from '../actions/user';
import initUiState from '../reducers/initUiState';
import * as errors from '../constants/errors';

import uiReducer from './ui';

describe('UI Reducer', () => {
  let initState;
  beforeEach(() => {
    initState = { ...initUiState };
  });

  it('initState when state is undefined', () => {
    expect(uiReducer(undefined, {})).toEqual(initState);
  });

  describe('UI Errors', () => {
    it('handles SET_UI_ERROR action', () => {
      const newState = {
        ...initState,
        error: errors.invalidEmail,
      };
      expect(
        uiReducer(initState, uiActions.setUiError(errors.invalidEmail)),
      ).toEqual(newState);
      expect(
        uiReducer(initState, uiActions.setUiError(errors.invalidEmail)),
      ).toMatchSnapshot();
    });

    it('RESET_UI_ERROR', () => {
      initState.error = errors.invalidEmail;
      const newState = {
        ...initState,
        error: {
          kind: '',
          header: '',
          message: '',
        },
      };
      expect(uiReducer(initState, uiActions.resetUiError())).toEqual(newState);
      expect(uiReducer(initState, uiActions.resetUiError())).toMatchSnapshot();
    });
  });

  describe('Boards Explorer', () => {
    it('handles SHOW_BOARDS_EXPLORER action', () => {
      const newState = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          boardsExplorer: true,
        },
      };
      expect(uiReducer(initState, uiActions.showBoardsExplorer())).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, uiActions.showBoardsExplorer()),
      ).toMatchSnapshot();
    });

    it('handles HIDE_BOARDS_EXPLORER action', () => {
      const state = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          boardsExplorer: true,
        },
      };
      expect(uiReducer(state, uiActions.hideBoardsExplorer())).toEqual(
        initState,
      );
      expect(
        uiReducer(state, uiActions.hideBoardsExplorer()),
      ).toMatchSnapshot();
    });
  });

  describe('Header Search', () => {
    it('handles FOCUS_HEADER_SEARCH action', () => {
      const newState = {
        ...initState,
        focus: 'search',
      };
      expect(uiReducer(initState, uiActions.focusHeaderSearch())).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, uiActions.focusHeaderSearch()),
      ).toMatchSnapshot();
    });

    it('handles BLUR_HEADER_SEARCH action', () => {
      const newState = {
        ...initState,
        focus: 'search',
        search: 'something',
      };
      expect(uiReducer(newState, uiActions.blurHeaderSearch())).toEqual(
        initState,
      );
      expect(
        uiReducer(newState, uiActions.blurHeaderSearch()),
      ).toMatchSnapshot();
    });

    it('handles SET_HEADER_SEARCH action', () => {
      const search = 'something';
      const newState = {
        ...initState,
        search,
      };
      expect(uiReducer(initState, uiActions.setHeaderSearch(search))).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, uiActions.setHeaderSearch(search)),
      ).toMatchSnapshot();
    });
  });

  describe('Create Menu', () => {
    it('handles SHOW_CREATE_MENU action', () => {
      const newState = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          create: true,
        },
      };
      expect(uiReducer(initState, uiActions.showCreateMenu())).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, uiActions.showCreateMenu()),
      ).toMatchSnapshot();
    });

    it('handles HIDE_CREATE_MENU action', () => {
      const state = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          create: true,
        },
      };
      expect(uiReducer(state, uiActions.hideCreateMenu())).toEqual(initState);
      expect(
        uiReducer(initState, uiActions.hideCreateMenu()),
      ).toMatchSnapshot();
    });
  });

  describe('Notifications', () => {
    it('handles SHOW_NOTIFICATIONS action', () => {
      const newState = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          notifications: true,
        },
      };
      expect(uiReducer(initState, uiActions.showNotifications())).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, uiActions.showNotifications()),
      ).toMatchSnapshot();
    });

    it('handles HIDE_NOTIFICATIONS action', () => {
      const state = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          notifications: true,
        },
      };
      expect(uiReducer(state, uiActions.hideNotifications())).toEqual(
        initState,
      );
      expect(
        uiReducer(initState, uiActions.hideNotifications()),
      ).toMatchSnapshot();
    });
  });

  describe('User Menu', () => {
    it('handles SHOW_USER_MENU action', () => {
      const newState = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          user: true,
        },
      };
      expect(uiReducer(initState, uiActions.showUserMenu())).toEqual(newState);
      expect(uiReducer(initState, uiActions.showUserMenu())).toMatchSnapshot();
    });

    it('handles HIDE_USER_MENU action', () => {
      const state = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          user: true,
        },
      };
      expect(uiReducer(state, uiActions.hideUserMenu())).toEqual(initState);
      expect(uiReducer(initState, uiActions.hideUserMenu())).toMatchSnapshot();
    });
  });

  describe('Rename Board', () => {
    it('handles SHOW_RENAME_BOARD action', () => {
      const newState = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          renameBoard: true,
        },
      };
      expect(uiReducer(initState, uiActions.showRenameBoard())).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, uiActions.showRenameBoard()),
      ).toMatchSnapshot();
    });

    it('handles HIDE_RENAME_BOARD action', () => {
      const state = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          renameBoard: true,
        },
      };
      expect(uiReducer(state, uiActions.hideRenameBoard())).toEqual(initState);
      expect(
        uiReducer(initState, uiActions.hideRenameBoard()),
      ).toMatchSnapshot();
    });
  });

  describe('Add to Team', () => {
    it('handles SHOW_ADD_TO_TEAM action', () => {
      const newState = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          addToTeam: true,
        },
      };
      expect(uiReducer(initState, uiActions.showAddToTeam())).toEqual(newState);
      expect(uiReducer(initState, uiActions.showAddToTeam())).toMatchSnapshot();
    });

    it('handles HIDE_ADD_TO_TEAM action', () => {
      const state = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          addToTeam: true,
        },
      };
      expect(uiReducer(state, uiActions.hideAddToTeam())).toEqual(initState);
      expect(uiReducer(initState, uiActions.hideAddToTeam())).toMatchSnapshot();
    });
  });

  describe('Visibility Menu', () => {
    it('handles SHOW_VISIBILITY_MENU action', () => {
      const newState = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          visibility: true,
        },
      };
      expect(uiReducer(initState, uiActions.showVisibilityMenu())).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, uiActions.showVisibilityMenu()),
      ).toMatchSnapshot();
    });

    it('handles HIDE_VISIBILITY_MENU action', () => {
      const state = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          visibility: true,
        },
      };
      expect(uiReducer(state, uiActions.hideVisibilityMenu())).toEqual(
        initState,
      );
      expect(
        uiReducer(initState, uiActions.hideVisibilityMenu()),
      ).toMatchSnapshot();
    });
  });

  describe('Board Menu', () => {
    it('handles SHOW_BOARD_MENU action', () => {
      const newState = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          board: true,
        },
      };
      expect(uiReducer(initState, uiActions.showBoardMenu())).toEqual(newState);
      expect(uiReducer(initState, uiActions.showBoardMenu())).toMatchSnapshot();
    });

    it('handles HIDE_BOARD_MENU action', () => {
      const state = {
        ...initState,
        activeMenus: {
          ...initState.activeMenus,
          board: true,
        },
      };
      expect(uiReducer(state, uiActions.hideBoardMenu())).toEqual(initState);
      expect(uiReducer(initState, uiActions.hideBoardMenu())).toMatchSnapshot();
    });
  });

  describe('Signup User', () => {
    it('handles SIGNUP_USER_REQUEST action', () => {
      const newState = {
        ...initState,
        fetching: {
          ...initState.fetching,
          signup: true,
        },
      };
      expect(uiReducer(initState, userActions.signupUserRequest())).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, userActions.signupUserRequest()),
      ).toMatchSnapshot();
    });

    it('handles SIGNUP_USER_SUCCESS action', () => {
      const newState = {
        ...initState,
        fetching: {
          ...initState.fetching,
          signup: false,
        },
      };
      expect(uiReducer(initState, userActions.signupUserSuccess())).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, userActions.signupUserSuccess()),
      ).toMatchSnapshot();
    });

    it('handles SIGNUP_USER_FAILURE action', () => {
      const newState = {
        ...initState,
        error: errors.serverError,
        fetching: {
          ...initState.fetching,
          signup: false,
        },
      };
      expect(
        uiReducer(initState, userActions.signupUserFailure(errors.serverError)),
      ).toEqual(newState);
      expect(
        uiReducer(initState, userActions.signupUserFailure(errors.serverError)),
      ).toMatchSnapshot();
    });
  });

  describe('Login User', () => {
    it('handles LOGIN_USER_REQUEST action', () => {
      const newState = {
        ...initState,
        fetching: {
          ...initState.fetching,
          login: true,
        },
      };
      expect(uiReducer(initState, userActions.loginUserRequest())).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, userActions.loginUserRequest()),
      ).toMatchSnapshot();
    });

    it('handles LOGIN_USER_SUCCESS action', () => {
      initState.isFetching = true;
      const newState = {
        ...initState,
        fetching: {
          ...initState.fetching,
          login: false,
        },
      };
      expect(uiReducer(initState, userActions.loginUserSuccess())).toEqual(
        newState,
      );
      expect(
        uiReducer(initState, userActions.loginUserSuccess()),
      ).toMatchSnapshot();
    });

    it('handles LOGIN_USER_FAILURE action', () => {
      initState.isFetching = true;
      const newState = {
        ...initState,
        error: errors.wrongCredentials,
        fetching: {
          ...initState.fetching,
          login: false,
        },
      };
      expect(
        uiReducer(
          initState,
          userActions.loginUserFailure(errors.wrongCredentials),
        ),
      ).toEqual(newState);
      expect(
        uiReducer(
          initState,
          userActions.loginUserFailure(errors.wrongCredentials),
        ),
      ).toMatchSnapshot();
    });
  });

  describe('Logout User', () => {
    it('handles LOGOUT_USER action', () => {
      initState.fetching.login = true;
      const newState = {
        ...initState,
        fetching: {
          ...initState.fetching,
          login: false,
        },
      };
      expect(uiReducer(initState, userActions.logoutUser())).toEqual(newState);
      expect(uiReducer(initState, userActions.logoutUser())).toMatchSnapshot();
    });
  });
});
