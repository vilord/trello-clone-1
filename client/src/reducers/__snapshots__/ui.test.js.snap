// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Reducer Add to Team handles HIDE_ADD_TO_TEAM action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Add to Team handles SHOW_ADD_TO_TEAM action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": true,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Board Menu handles HIDE_BOARD_MENU action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Board Menu handles SHOW_BOARD_MENU action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": true,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Boards Explorer handles HIDE_BOARDS_EXPLORER action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Boards Explorer handles SHOW_BOARDS_EXPLORER action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": true,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Create Menu handles HIDE_CREATE_MENU action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Create Menu handles SHOW_CREATE_MENU action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": true,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Header Search handles BLUR_HEADER_SEARCH action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Header Search handles FOCUS_HEADER_SEARCH action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "search",
  "search": "",
}
`;

exports[`UI Reducer Header Search handles SET_HEADER_SEARCH action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "something",
}
`;

exports[`UI Reducer Login User handles LOGIN_USER_FAILURE action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "Something wrong!",
    "kind": "wrongCredentials",
    "message": "Incorrect email (username) or password.",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "isFetching": true,
  "search": "",
}
`;

exports[`UI Reducer Login User handles LOGIN_USER_REQUEST action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": true,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Login User handles LOGIN_USER_SUCCESS action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "isFetching": true,
  "search": "",
}
`;

exports[`UI Reducer Logout User handles LOGOUT_USER action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Notifications handles HIDE_NOTIFICATIONS action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Notifications handles SHOW_NOTIFICATIONS action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": true,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Rename Board handles HIDE_RENAME_BOARD action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Rename Board handles SHOW_RENAME_BOARD action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": true,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Signup User handles SIGNUP_USER_FAILURE action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "Unknown Server Error!",
    "kind": "serverError",
    "message": "Our server may be down. Please retry later!",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Signup User handles SIGNUP_USER_REQUEST action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": true,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Signup User handles SIGNUP_USER_SUCCESS action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer UI Errors RESET_UI_ERROR 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer UI Errors handles SET_UI_ERROR action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "Verify email!",
    "kind": "invalidEmail",
    "message": "That doesn't look like an email address…",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer User Menu handles HIDE_USER_MENU action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer User Menu handles SHOW_USER_MENU action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": true,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Visibility Menu handles HIDE_VISIBILITY_MENU action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": false,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;

exports[`UI Reducer Visibility Menu handles SHOW_VISIBILITY_MENU action 1`] = `
Object {
  "activeMenus": Object {
    "addToTeam": false,
    "board": false,
    "boardsExplorer": false,
    "create": false,
    "notifications": false,
    "renameBoard": false,
    "user": false,
    "visibility": true,
  },
  "error": Object {
    "header": "",
    "kind": "",
    "message": "",
  },
  "fetching": Object {
    "activity": false,
    "login": false,
    "signup": false,
  },
  "focus": "",
  "search": "",
}
`;
